src/utils/daemon.rs

```diff
@@ -4,13 +4,62 @@
 
 use crate::{
     utils::{queue_builder, report_error},
-    Context, RustwideBuilder,
+    BuildQueue, Config, Context, Index, RustwideBuilder,
 };
 use anyhow::{anyhow, Context as _, Error};
 use log::{debug, info};
+use std::sync::Arc;
 use std::thread;
 use std::time::{Duration, Instant};
 
+/// Run the registry watcher
+/// NOTE: this should only be run once, otherwise crates would be added
+/// to the queue multiple times.
+pub fn watch_registry(
+    build_queue: Arc<BuildQueue>,
+    config: Arc<Config>,
+    index: Arc<Index>,
+) -> Result<(), Error> {
+    let mut last_gc = Instant::now();
+
+    // On startup we fetch the last seen index reference from
+    // the database and set it in the local index repository.
+    match build_queue.last_seen_reference() {
+        Ok(Some(oid)) => {
+            index.diff()?.set_last_seen_reference(oid)?;
+        }
+        Ok(None) => {}
+        Err(err) => {
+            log::error!(
+                "queue locked because of invalid last_seen_index_reference in database: {}",
+                err
+            );
```

Comment:
We should use `{:?}` to print anyhow errors [to include the cause and backtrace if available](https://docs.rs/anyhow/latest/anyhow/struct.Error.html#display-representations) (and can use [captured identifiers](https://blog.rust-lang.org/2022/01/13/Rust-1.58.0.html#captured-identifiers-in-format-strings)).

```suggestion
            log::error!(
                "queue locked because of invalid last_seen_index_reference in database: {err:?}",
            );
```

Reply:
good catch! I typically do this (`{:?}`), this time I missed it.

I fixed it.